### PR TITLE
[gardening] Remove old unavailable APIs

### PR DESCRIPTION
**Pull request by ikesyo**
#### Checklist
- ~~[ ] Updated CHANGELOG.md.~~

### CODE REVIEW

File: Sources/Deprecations+Removals.swift [hunk 1]
```diff
@@ -2,200 +2,10 @@ import Foundation
...
-
-@available(*, unavailable, renamed:"Scheduler")
-public typealias SchedulerProtocol = Scheduler
-
-@available(*, unavailable, renamed:"DateScheduler")
-public typealias DateSchedulerProtocol = DateScheduler
-
-@available(*, unavailable, renamed:"BindingSource")
-public typealias BindingSourceProtocol = BindingSource
+// MARK: Deprecated types in ReactiveSwift 2.x.
```
Comment (mdiep) [inline Sources/Deprecations+Removals.swift]: This probably shouldn't say `2.x`?

File: Sources/Deprecations+Removals.swift [hunk 2]
```diff
@@ -2,200 +2,10 @@ import Foundation
...
-	@discardableResult
-	@available(*, unavailable, renamed:"observeEnded(_:)")
-	public func add(_ action: () -> Void) -> Disposable? { fatalError() }
-
-	@discardableResult
-	@available(*, unavailable, message:"Use `observeEnded(_:)` instead.")
-	public static func += (left: Lifetime, right: () -> Void) -> Disposable? { fatalError() }
-
-	@discardableResult
-	@available(*, deprecated, message:"Use `observeEnded(_:)` with a method reference to `dispose()` instead. This method is subject to removal in a future release.")
```
Comment (mdiep) [inline Sources/Deprecations+Removals.swift]: Since this one was only `deprecated`, it should stay until the next major version and become `unavailable`.

**Comment by mdiep** [comment]
Thanks for cleaning this out. ✨

File: Sources/Deprecations+Removals.swift (hunk 2 above)
Comment (ikesyo) [reply Sources/Deprecations+Removals.swift]: Oh overlooked that 🙈

Comment (ikesyo) [reply Sources/Deprecations+Removals.swift]: Addressed in 0a378cb

**Pull request merged**